src/modules/home/components/accommodation-list.tsx

```diff
@@ -0,0 +1,62 @@
+import Grid2 from '@mui/material/Grid2';
+import CardImage from '#/assets/images/card-temp-image.jpg';
+import { AccommodationCard } from './accommodation-card';
+import { useListAccommodationsQuery } from '../api/list-accommodations.api';
+import { useEffect, useState } from 'react';
+import { type Accommodation } from '../types/accommodation.type';
+import { SkeletonList } from './skeleton-list';
+
+export function AccommodationList() {
+  const [accommodations, setAccommodations] = useState<Accommodation[]>([]);
+  const [page, setPage] = useState(1);
+  const [loading, setLoading] = useState(false);
+  const [initialLoading, setInitialLoading] = useState(true);
+
+  const limit = 20;
+  const { error, data: fetchedAccommodations } = useListAccommodationsQuery(page, limit);
+
+  useEffect(() => {
+    if (fetchedAccommodations) {
+      setAccommodations((prev) => [...prev, ...fetchedAccommodations]);
+      setLoading(false);
+      setInitialLoading(false);
+    }
+  }, [fetchedAccommodations]);
+
+  useEffect(() => {
+    const handleScroll = () => {
```

Comment:
I found that this component is re-rendering to many times when scrolling, maybe it's better to use debounce for handleScroll